pyproject.toml

```diff
@@ -2,7 +2,7 @@
 build-backend = "mesonpy"
 requires = [
     "Cython>=0.29.34,<3.1",
-    "meson-python>=0.15.0",
+    "meson-python>=0.15.0,<0.16.0",
```

Comment:
This seems fine to me, but just to make sure: this is preventative only right, there was no regression?

Reply:
The macos arm64 wheel builds were failing. Homebrew changed install directories and it seems meson-python 0.16.0 (released recently) was also failing. Not sure why.

Reply:
There error seems to be:
```
  ../meson.build:1:0: ERROR: Unknown compiler(s): [['cc'], ['gcc'], ['clang'], ['nvc'], ['pgcc'], ['icc'], ['icx']]
  The following exception(s) were encountered:
  Running `nvc --version` gave "[Errno 2] No such file or directory: 'nvc'"
  Running `pgcc --version` gave "[Errno 2] No such file or directory: 'pgcc'"
  Running `icc --version` gave "[Errno 2] No such file or directory: 'icc'"
  Running `icx --version` gave "[Errno 2] No such file or directory: 'icx'"
  A full log can be found at /private/var/folders/1z/65jd_97j40z81zjth6w6qbs40000gn/T/cirrus-ci-build/.mesonpy-7gwbi_k9/meson-logs/meson-log.txt
  error: subprocess-exited-with-error
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
```

Reply:
I note that cibuildwheel was not pinned in these builds, which might be related.

Reply:
@rgommers Any idea where those errors come from?

Reply:
That's almost certainly only due to a change in the runner image. meson-python doesn't influence C compiler selection in any way, and I've never seen it go wrong like this on macOS since the system compilers are normally always found.